libvast/include/vast/bitmap_algorithms.hpp

```diff
@@ -492,6 +508,29 @@ auto select(const IDs& ids) {
   return select_range<Bit, range_type>(bit_range(ids));
 }
 
+/// Yields all contiguous ranges of *Bit* in *bitmap*.
+/// @param bitmap The bitmap.
+/// @returns A generator id ranges.
+template <bool Bit = true, class Bitmap>
+auto select_runs(const Bitmap& bitmap) -> detail::generator<id_range> {
```

Comment:
I wish I had generators when I wrote the first version of all these algorithms. So much complexity now...

Reply:
I thought the same when I wrote this—this was so easy to write. I would love to just scrap the bitmap stuff and to reinvent it with generators. There are so many opportunities for optimization here.